cupy/core/_routines_math.pyx

```diff
@@ -424,8 +528,13 @@ cdef ndarray scan(ndarray a, op, dtype=None, ndarray out=None):
 
     cdef int src_cont = int(a._c_contiguous)
     cdef int out_cont = int(out._c_contiguous)
-    kern_scan = _inclusive_scan_kernel(a.dtype, dtype, block_size, op,
-                                       src_cont, out_cont)
+    if (numpy.dtype(dtype).char in 'iIlLqQfd' and
+            not runtime._is_hip_environment):
```

Comment:
Do we also need a CC check here?

Reply:
Looks like warp shuffles are already available as early as CC 3.x, so I'd say it's fine?
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-shuffle-functions

Reply:
Yeah, cuda 9.x supports only from 3.0 so no problem here! thanks :)

Reply:
I think that we do not need to add a CC checking here as warp shuffle functions are supported in CC 3.x or higher.
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-shuffle-functions